common/ferrostar/src/algorithms.rs

```diff
@@ -273,28 +273,33 @@ fn distance_to_end_of_step(snapped_location: &Point, current_step_linestring: &L
 
 /// Computes the arrival state for a snapped location along the route.
 /// This includes distances and durations.
+///
+/// NOTE: `remaining_steps` includes the current step.
 pub fn calculate_trip_progress(
     snapped_location: &Point,
-    current_step: &RouteStep,
     current_step_linestring: &LineString,
     remaining_steps: &[RouteStep],
 ) -> TripProgress {
-    if remaining_steps.is_empty() {
+    let Some(current_step) = remaining_steps.first() else {
```

Comment:
Passing current_step explicitly here opened the door for potentially very confusing errors caused by this ambiguity. I've just made it impossible to make that mistake now.